### PR TITLE
Pchain UTs cleanup

### PR DESCRIPTION
**Pull request by abi87**
## Why this should be merged
platformVM unit tests needs to be easier to maintain.
Specifically I currently find non-trivial to understand what specific fork a unit test is testing and possibly to update the tests to the latest fork.
Also fork times do not make sense and vm clock is not set to fullfil them.

## How this works
Just refactor tests so that the latest active fork is explictly set upon vm creation and vm.clock is set to make the fork active.

## How this was tested
Testing test is tricky. I guess I tested by manual inspection

### CODE REVIEW

File: vms/platformvm/block/builder/helpers_test.go [hunk 1]
```diff
@@ -303,16 +348,17 @@ func defaultConfig() *config.Config {
...
 	}
 }
 
 func defaultClock() *mockable.Clock {
-	// set time after Banff fork (and before default nextStakerTime)
+	now := latestForkTime
 	clk := &mockable.Clock{}
-	clk.Set(defaultGenesisTime)
+	clk.Set(now)
 	return clk
```
Comment (anusha-ctrl) [inline vms/platformvm/block/builder/helpers_test.go:362]: is it cleaner to just say `clk.Set(latestForkTime)`

File: vms/platformvm/block/executor/verifier_test.go [hunk 2]
```diff
@@ -29,10 +29,15 @@ import (
...
 )
 
-func TestVerifierVisitProposalBlock(t *testing.T) {
+func TestVerifierVisitApricotProposalBlock(t *testing.T) {
 	require := require.New(t)
 	ctrl := gomock.NewController(t)
 
+	config := &config.Config{
+		BanffTime:   mockable.MaxTime, // banff is not activated
+		CortinaTime: mockable.MaxTime, // cortina is not activate
```
Comment (anusha-ctrl) [inline vms/platformvm/block/executor/verifier_test.go:38]:
```suggestion
		CortinaTime: mockable.MaxTime, // cortina is not activated
```

Comment (anusha-ctrl) [reply vms/platformvm/block/executor/verifier_test.go:38]: same for below

Comment (anusha-ctrl) [reply vms/platformvm/block/executor/verifier_test.go:38]: it might be worth creating a `notActivatedConfig`

File: vms/platformvm/state/staker_test.go [hunk 3]
```diff
@@ -143,8 +143,8 @@ func TestNewCurrentStaker(t *testing.T) {
 	publicKey := bls.PublicFromSecretKey(sk)
 	subnetID := ids.GenerateTestID()
 	weight := uint64(12345)
-	startTime := time.Now()
-	endTime := time.Now()
+	startTime := time.Now().Truncate(time.Second)
+	endTime := time.Now().Truncate(time.Second)
 	potentialReward := uint64(54321)
```
Comment (anusha-ctrl) [inline vms/platformvm/state/staker_test.go:148]: Why was this changed?

File: vms/platformvm/block/builder/helpers_test.go (hunk 1 above)
Comment (anusha-ctrl) [inline vms/platformvm/block/builder/helpers_test.go:362]:
```suggestion
func defaultClock() *mockable.Clock {
clock := &mockable.Clock{}
clock.Set(latestForkTime)
return clock
}
```

**Comment by abi87** [comment]
Replaced by https://github.com/ava-labs/avalanchego/pull/2169. Thanks @anusha-ctrl for the comments that pushed me to revise this PR (and replaced it).
